src/basic/virt.c

```diff
@@ -472,8 +465,18 @@ int detect_vm(void) {
                 return r;
         if (r == VIRTUALIZATION_VM_OTHER)
                 other = true;
-        else if (r != VIRTUALIZATION_NONE)
-                goto finish;
+        else if (r != VIRTUALIZATION_NONE) {
+                 /* If we are Dom0, then we expect to not report as a VM. However, as we might be nested
+                  * inside another hypervisor which can be detected via the CPUID check, wait to report this
+                  * until after the CPUID check. */
+                xen_dom0 = detect_vm_xen_dom0();
+                if (xen_dom0 < 0)
+                        return r;
```

Comment:
I figure you mean `return xen_dom0` here?

Reply:
Correct - thanks!